statistics/histogram.go

```diff
@@ -43,26 +46,113 @@ type Histogram struct {
 	// LastUpdateVersion is the version that this histogram updated last time.
 	LastUpdateVersion uint64
 
-	Buckets []Bucket
+	tp *types.FieldType
+
+	// Histogram elements.
+	//
+	// A bucket bound is the smallest and greatest values stored in the bucket. The lower and upper bound
+	// are stored in one column.
+	//
+	// A bucket count is the number of items stored in all previous buckets and the current bucket.
+	// Bucket counts are always in increasing order.
+	//
+	// Repeat is the number of repeats of the bucket value, it can be used to find popular values.
+	Bounds  *chunk.Chunk
+	Counts  []int64
```

Comment:
If we still use a struct `Bucket` to store `counts`, `repeats`, `lowerScalar`, `upperScaler`, `commonPfxLen`, it will save more memory.

Reply:
For some simple type like `Int`, we do not need to store `lowerScalar`, `upperScalar`, `commonPfxLen`, I think it will save more.

Reply:
Then we can use two structs, one for `Counts` and `Repeats`, another one for `lowerScalar`,`upperScalar`, `commonPfxLen`.

Reply:
But for type `Decimal` and `Time`, we do not need to store the `commonPfxLen`...